.docker/docker_start_script.sh

```diff
@@ -7,19 +7,17 @@
 echo ${DB_HOST}
 if [ -z ${DB_HOST} ] ; then
     local_host=`hostname -i`
-    jq --arg db_host "$local_host" '.timeseries.url = $db_host' conf/storage/db.conf.sample > conf/storage/db.conf
-else
-    jq --arg db_host "$DB_HOST" '.timeseries.url = $db_host' conf/storage/db.conf.sample > conf/storage/db.conf
+    export DB_HOST=$local_host
+    echo "Setting db host environment variable to localhost"
```

Comment:
@shankari You're right, it does not work without DB_HOST set. I ran this:
```
docker run --name em-server-no-host -d -e WEB_SERVER_HOST=0.0.0.0 -e STUDY_CONFIG=stage-program --network emission consolidate-diff-server-localbuild
```
and before I could exec into the container, it gave me this error: 

```
2024-05-13 15:28:05 pymongo.errors.ServerSelectionTimeoutError: 172.26.0.3:27017: [Errno 111] Connection refused, Timeout: 30s, Topology Description: <TopologyDescription id: 664285c63b13fd554aa58ca9, topology_type: Unknown, servers: [<ServerDescription ('172.26.0.3', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('172.26.0.3:27017: [Errno 111] Connection refused')>]>
```

Reply:
@shankari I've tried several different ways to export the DB_HOST environment variable, but kept getting the `connection refused` error. I decided to try running the server container **as it currently is in production** by building a local image with an added echo.

```
docker build -t localhost-5 .
docker run --name em-server-no-host -d -e WEB_SERVER_HOST=0.0.0.0 -e STUDY_CONFIG=stage-program --network emission localhost-5
```
The echo I added is in docker_start_script.sh, after it's supposedly set with the sample file:

```
echo ${DB_HOST}
if [ -z ${DB_HOST} ] ; then
    local_host=`hostname -i`
    jq --arg db_host "$local_host" '.timeseries.url = $db_host' conf/storage/db.conf.sample > conf/storage/db.conf
else
    jq --arg db_host "$DB_HOST" '.timeseries.url = $db_host' conf/storage/db.conf.sample > conf/storage/db.conf
fi
cat conf/storage/db.conf

echo "LOCALHOST!!:" ${DB_HOST}
```

Container log:
```
2024-05-14 12:55:08 {
2024-05-14 12:55:08   "timeseries": {
2024-05-14 12:55:08     "url": "172.26.0.3",
2024-05-14 12:55:08     "result_limit": 250000
2024-05-14 12:55:08   }
2024-05-14 12:55:08 }
2024-05-14 12:55:08 LOCALHOST!!:
```
I got the connection refused error again. Thoughts: 
* It isn't working in the first place
* I'm misunderstanding what's happening
* My db container is the problem
Going to try a new db container first and see if that's the issue.

Reply:
Tried new DB container:
`docker run --name db -d -p 27017:27017 --network emission mongo:4.4.0` 

Ran the exact same image as before, and got the same connection refused error.

Reply:
I don't understand the challenge here.
https://github.com/e-mission/e-mission-server/pull/961#discussion_r1600524723

The previous codebase assumed that if the DB_HOST was not set, the DB and the webapp were running on the same host. Obviously, if you try to test it with containers, it won't work.

However, with the new codebase, even if you run the DB and the webapp on the same container, it won't work because of https://github.com/e-mission/e-mission-server/pull/961#discussion_r1597355336

Reply:
Per our meeting, we clarified that:

* The `else` was removed, so this code should work as expected
* This method of setting the variable doesn't work when using containers, explaining the failures in my tests above

Reply:
If this doesn't work with containers, then do we even need this functionality in a file that only runs when the dockerfile is run?? It makes sense in start_script.sh, but maybe not here.